nixos/modules/services/monitoring/prometheus/exporters/mktxp.nix

```diff
@@ -0,0 +1,586 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    getExe
+    mkPackageOption
+    mkOption
+    types
+    escapeShellArgs
+    concatStringsSep
+    ;
+  inherit (lib.generators) mkKeyValueDefault mkValueStringDefault;
+  inherit (lib.attrsets) recursiveUpdate;
+
+  cfg = config.services.prometheus.exporters.mktxp;
+
+  settingsFormat = pkgs.formats.ini {
+    mkKeyValue = mkKeyValueDefault rec {
+      mkValueString =
+        v:
+        if builtins.isNull v then
+          "None"
+        else if builtins.isBool v then
+          (if v then "True" else "False")
+        else if builtins.isList v then
+          # Join lists as "a, b, c"
+          concatStringsSep ", " (map mkValueString v)
+        else
+          mkValueStringDefault { } v;
+    } "=";
+  };
+
+  sectionType = settingsFormat.lib.types.atom; # attrsOf INI atom
+
+  configFile =
+    if cfg.configFile != null then
+      cfg.configFile
+    else
+      let
+        mktxpSettings = {
+          MKTXP.listen = "${cfg.listenAddress}:${toString cfg.port}";
+        };
+        merged = recursiveUpdate mktxpSettings cfg.settings;
+      in
+      settingsFormat.generate "mktxp.conf" merged;
+in
+{
+  port = 49090;
+
+  extraOpts = {
+    package = mkPackageOption pkgs "mktxp" { };
+
+    configFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        The path to the mktxp configuration ini file. For more info see
+        https://github.com/akpw/mktxp
+
+        Note that setting this option is mutually exclusive with setting
+        the `settings` option.
+      '';
+    };
+
+    settings =
```

Comment:
It seems completely overkill to declare and slightly painful to maintain all of these options here. I imagine it could easily lead to a lot of renames/removed option imports over time.

Is it the case that the rendered config file needs to have all of these options in order for the program to accept it? If so, could you instead limit the options to those you know people want to change and/or those with weird type conversions, and then just add the rest of the defaults via the `recursiveUpdate` further up?